source/common/filter/tcp_proxy.cc

```diff
@@ -324,13 +330,15 @@ void TcpProxy::onDownstreamEvent(Network::ConnectionEvent event) {
     // downstream connection to stick around, or, we need to be able to pass this connection to a
     // flush worker which will attempt to flush the remaining data with a timeout.
     upstream_connection_->close(Network::ConnectionCloseType::NoFlush);
+    idle_timer_.reset();
```

Comment:
It's idiomatic in the rest of the code to call `disable()` prior to reset only so that you can EXPECT the disable call in tests (the `onDestroy()` pattern in destructor does not work with NiceMock which would create a mess for timers in general which is why we don't do that in all cases). You might consider doing that here and below and then updating the tests.